R/plotSpectra-PTM.R

```diff
@@ -0,0 +1,482 @@
+#' @title Function to plot MS/MS spectra with PTMs
+#'
+#' @description
+#' `plotSpectraPTM()` creates annotated visualisations of MS/MS spectra,
+#' designed to explore fragment identifications and post-translational
+#' modifications (PTMs).
+#' 
+#' `plotSpectraPTM()` plots a spectrum's m/z values on the x-axis and corresponding 
+#' intensities on the y-axis, labeling the peaks according to theoretical 
+#' fragment ions (e.g., b, y, a, c, x, z) computed using `labelFragments()`
+#' and `calculateFragments()`.
+#'
+#' @param x a `Spectra()` object.
+#' 
+#' @param deltaMz `logical(1L)` If `TRUE`, adds an additional plot showing the 
+#' difference of mass over charge between matched oberved and theoretical 
+#' fragments in parts per million. Does not yet support modifications. The 
+#' matching is based on `calculateFragments()` and needs a 'sequence' variable
+#' in `spectraVariables(x)`. Default is set to `TRUE`.
+#'
+#' @param ppm `integer(1L)` Sets the limits of the delta m/z plot and is passed
+#' to `labelFragments()`.
+#'
+#' @param xlab `character(1)` with the label for the x-axis 
+#' (by default `xlab = "m/z"`).
+#'
+#' @param ylab `character(1)` with the label for the y-axis 
+#' (by default `ylab = "intensity"`).
+#'
+#' @param xlim `numeric(2)` defining the x-axis limits. The range of m/z values 
+#' are used by default.
+#'
+#' @param ylim `numeric(2)` defining the y-axis limits. The range of intensity 
+#' values are used by default.
+#'
+#' @param main `character(1)` with the title for the plot. By default the 
+#' spectrum's MS level and retention time (in seconds) is used.
+#'
+#' @param col Named `character(4L)`. Colors for the labels, the character names
+#' need to be "b", "y", "acxz" and "other", respectively for the b-ions, y-ions,
+#' a,c,x,z-ions and the unidentified fragments. 
+#'
+#' @param labelCex `numeric(1)` giving the amount by which the text should be
+#' magnified relative to the default. See parameter `cex` in `par()`.
+#'
+#' @param labelSrt `numeric(1)` defining the rotation of the label. 
+#' See parameter `srt` in `text()`.
+#'
+#' @param labelAdj see parameter `adj` in `text()`.
+#'
+#' @param labelPos see parameter `pos` in `text()`.
+#'
+#' @param labelOffset see parameter `offset` in `text()`.
+#'
+#' @param asp for `plotSpectra()`: the target ratio (columns / rows) when plotting
+#' mutliple spectra (e.g. for 20 spectra use asp = 4/5 for 4 columns and 5 rows
+#' or asp = 5/4 for 5 columns and 4 rows; see `grDevices::n2mfrow()` for details).
+#' If `deltaMz` is `TRUE`, `asp` is ignored.
+#' 
+#' @param minorTicks `logical(1L)` If `TRUE`, minor ticks are added to the plots.
+#' Default is set to `TRUE`.
+#'   
+#' @param ... additional parameters to be passed to the `labelFragments()` function.
+#' 
+#' @importFrom Spectra spectraVariables
+#'
+#' @author Johannes Rainer, Sebastian Gibb, Guillaume Deflandre, Laurent Gatto
+#' 
+#' @seealso \code{\link[Spectra]{plotSpectra}}
```

Comment:
Please use markdown syntax [Spectra::plotSpectra()] (see [here](https://roxygen2.r-lib.org/articles/markdown.html#links)) in all roxygen headers.